src/steadystateproblem.cpp

```diff
@@ -23,8 +23,13 @@ SteadystateProblem::SteadystateProblem(const Solver &solver, const Model &model)
       x(model.nx_solver), x_old(model.nx_solver), dx(model.nx_solver),
       xdot(model.nx_solver), xdot_old(model.nx_solver),
       sx(model.nx_solver, model.nplist()), sdx(model.nx_solver, model.nplist()),
+      maxSteps(std::max(solver.getNewtonMaxSteps(), 1)),
       dJydx(model.nJ * model.nx_solver * model.nt(), 0.0), numsteps(3, 0),
-      numlinsteps(solver.getNewtonMaxSteps() * 2, 0) {}
+      numlinsteps(2, 0) {
+          /* maxSteps must be adapted if iterative linear solvers are used */
+          if (solver.getLinearSolver() == LinearSolver::SPBCG)
+              numlinsteps.resize(2 * solver.getNewtonMaxSteps(), 0);
```

Comment:
This doesn't handle the 0 in solver.getNewtonMaxSteps

Reply:
Technically it's valid, although only possible when misspecified by the user... 
Yet: Cleaner to catch it here, I think...